src/radical/pilot/agent/lm/prte.py

```diff
@@ -81,41 +96,38 @@ def lrms_config_hook(cls, name, cfg, lrms, logger, profiler):
 
         # Split up the debug strings into args and add them to the cmdline
         cmdline += ' '.join(debug_strings)
+        cmdline  = cmdline.strip()
 
         vm_size = len(lrms.node_list)
-        logger.info("Start prte on %d nodes ['%s']", vm_size, ' '.join(cmdline))
+        logger.info("Start prte on %d nodes [%s]", vm_size, cmdline)
         profiler.prof(event='dvm_start', uid=cfg['pilot_id'])
 
         dvm_uri     = None
-        dvm_process = mp.Popen(cmdline.split(), stdout=mp.PIPE, stderr=mp.STDOUT)
-
-        while True:
+        dvm_process = mp.Popen(cmdline.split(), stdout=mp.PIPE,
+                               stderr=mp.STDOUT)
 
-            line = dvm_process.stdout.readline()
+        for _ in range(100):
 
-            if '://' in line:
-
-                dvm_uri = line.strip()
-                logger.info("prte uri: %s" % dvm_uri)
+            time.sleep(0.1)
+            try:
+                with open(furi, 'r') as fin:
+                    for line in fin.readlines():
+                        if '://' in line:
+                            dvm_uri = line.strip()
+                            break
```

Comment:
This is ugly :) Is there a way to get notified/pushed instead of trying 100 times and then assume sleep a bit more if nothing has happened?

Reply:
This is ugly, but unfortunately more reliable than parsing the (buffered) output stream of the HNP process.  The latter also reveals the URL eventually, but specifically if debug settings are enabled, the URL is well hidden in lots of noisy output, and we would need a more clever parser.

In principle we should be able to wait for a `DVM ready` message on the HNP output stream and should then expect to just find a valid URL in that file.  That actually was implemented before.  Alas, that also turned out to be not very reliable :-/

Summary: yes, ugly, but I don't see a better way right now.

Reply:
Thanks. Maybe something to escalate to the PRTE developers?